### PR TITLE
Add macOS Rendering support. Close #695

### LINKED ISSUE #695: macOS rendering support?
It seems QR code visualization on the Mac is not being supported. Would be great to get this working via Xamarin.Mac.

### PR DESCRIPTION
**Pull request by EBrown8534**
This will add rendering support for macOS clients.

There is also a sample that shows the use of it. It's extremely similar to iOS.

### CODE REVIEW

**Comment by knocte** [comment]
FYI the reason of why AppVeyor build fails might be https://github.com/Redth/ZXing.Net.Mobile/issues/772 , which I fixed in https://github.com/Redth/ZXing.Net.Mobile/pull/774 .

File: ZXing.Net.Mobile.Forms.sln [hunk 1]
```diff
@@ -29,6 +29,8 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "ZXing.Net.Mobile.Forms.Wind
 EndProject
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "ZXing.Net.Mobile.Forms.iOS", "Source\ZXing.Net.Mobile.Forms.iOS\ZXing.Net.Mobile.Forms.iOS.csproj", "{8370ADB4-229C-4DEC-908F-B1A8FC155523}"
 EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "ZXing.Net.Mobile.macOS", "Source\ZXing.Net.Mobile.macOS\ZXing.Net.Mobile.macOS.csproj", "{A7E26AE0-12AF-4CFD-9B1E-3101347CC6F4}"
+EndProject
```
Comment (knocte) [inline ZXing.Net.Mobile.Forms.sln]: maybe another new project ZXing.Net.Mobile.Forms.macOS is needed, to get Xamarin.Forms support?

Comment (EBrown8534) [reply ZXing.Net.Mobile.Forms.sln]: @knocte That would make sense, it seems like they define two classes: `ZXingScannerViewRenderer` and `ZXingBarcodeImageViewRenderer`, the former looks to be used by the scanner itself, while the latter looks to be used to render barcodes.

I'll get that thrown in here for macOS as well.

**Comment by knocte** [comment]
I think I've found a nitpick that would be great to include in this PR: adding the corresponding macOS bits to the nuspec files. Otherwise this wouldn't work automatically when including it via Nuget in a Xamarin.Forms project, whenever a new nuget version is published.

**Pull request merged**